Get Latest Raven Updates

tgroshon opened the pull request:
For the following:
- Better express 4 support
- Node 0.12 and IO.js support

stambata committed:
Update utils.js

Please consider the proposed change.The source code of the non-native stacktrace lines was never fetched on Windows machines. That's because isInternal was always resolving to true no matter what:

```js

isInternal = line.isNative() || (frame.filename[0] !== '/' && frame.filename[0] !== '.'); 

```

I added 2 small modifications:
1) the hardcoded separator ('/') has been replaced with path.sep (please refer to: https://nodejs.org/api/path.html#path_path_sep). That way the check is valid for all platforms: '/' for linux and '\\' for windows
2) an extra check has been added: frame.filename.indexOf(':\\') != 1
That way absolute paths on windows are considered non-native.

stambata committed:
Update utils.js

stambata committed:
Update utils.js

stambata committed:
Update raven.utils.js

FennZheng committed:
add maxReqQueueCount in case sentry server is down

benvinegar committed:
Fix formatting issues in docs (#184)

benvinegar committed:
Fix environment not attached from opts/kwargs (#185)

cc @macqueen @fboutin-pmc

benvinegar committed:
0.12.1

LewisJEllis committed:
Fix DSN template placeholder in express integration docs (#191)

mattrobenolt committed:
rm -rf udp (#197)

* rm -rf udp

This has been gone for years

LewisJEllis committed:
Update eslint to 2.13.1, make eslintrc more strict (#205)

LewisJEllis committed:
Update and lock down dependency versions (#206)

LewisJEllis committed:
Add client.runWithContext, make middleware use runWithContext, make runWithContext and set*Context play together

LewisJEllis committed:
Move toward raven-js API compatibility, make set*Context methods more BC

LewisJEllis committed:
Add alerting for deprecation/config/transmission warnings; fix #159

LewisJEllis committed:
Make wrap better, flesh out context family methods

LewisJEllis committed:
Make tests work

LewisJEllis committed:
Make capture* callbacks fire after sending, more API cleanup/singletonization

LewisJEllis committed:
Add node v6 to travis and change 4.0, 5.0 to 4.x, 5.x

LewisJEllis committed:
Still call process cb when sending is disabled

LewisJEllis committed:
Call cb in transport send on req error

LewisJEllis committed:
Add unhandledRejection handler

LewisJEllis committed:
Add shouldSendCallback, set*Callback methods

LewisJEllis committed:
Add test that callback fires after sending/receiving response

LewisJEllis committed:
Add tests for multi-callback chaining

LewisJEllis committed:
Remove node v0.8 from travis for nock compat

LewisJEllis committed:
Make unhandledRejection listener optional via .install param

LewisJEllis committed:
Add tests for unhandledRejection listeners

zwily commented:
I may be misunderstanding how pull requests work... How does this PR keep getting updated by people other than the original submitter?

LewisJEllis committed:
Add Node v7 to travis (#221)

LewisJEllis committed:
Middleware fixes (#223)

* Disable client instantiation in errorHandler

* Use captureException the new way

* Move middleware to live on Raven prototype

LewisJEllis committed:
Cleaning up wrap, adding interceptErr (#225)

* Cleaning up wrap, adding intercept

* Make intercept preserve function details like wrap

* Add intercept tests

* Change intercept to interceptErr

LewisJEllis committed:
Simplify context methods to fall back on instance-global context (#226)

* Simplify context methods to fall back on instance-global context

* Fix setContext, add merging hierarchy w/tests

* Simplify errorHandler middleware, call next immediately instead of waiting for capture

* Deprecate setUser/Tags/ExtraContext

* Rename/fix updateContext -> mergeContext

LewisJEllis committed:
Move set*Context to deprecations, fix captureError return value (#227)

LewisJEllis committed:
Version 1.0.0-beta.0 (#229)

* Version 1.0.0-beta.0

* Make version validity regex understand beta version naming scheme

adamchainz committed:
History - fix issue number for process.env (#230)

#181 not #182

Commit:
Add column number to Sentry payload (#235)

LewisJEllis committed:
WIP

LewisJEllis committed:
Check that require.main.filename is defined (#233)

scttcper committed:
node-uuid -> uuid (#236)

LewisJEllis committed:
Postgres autobreadcrumbing

LewisJEllis committed:
Make sure cb exists before calling it in should-not-send case (#238)

LewisJEllis committed:
Version 1.0.0-beta.1

LewisJEllis committed:
Refactored autobreadcrumb hooks, added per-breadcrumb-type config options

LewisJEllis committed:
Don't capture http breadcrumb for our own requests

LewisJEllis committed:
Deprecate/remove captureQuery (#239)

LewisJEllis committed:
Improve autoBreadcrumb configuration, make instrumentation defensive against missing modules

LewisJEllis committed:
Fix recursively nested values issue with breadcrumb references

LewisJEllis committed:
Add installed flag to avoid capturing console breadcrumbs prematurely

LewisJEllis committed:
Remove parseQuery

LewisJEllis committed:
Be defensive against missing domain.sentryContext

LewisJEllis committed:
Add data handler to instrumented responses to avoid non-consumption timeout issue

LewisJEllis committed:
Add instrumentation tests for node-http against its own tests

LewisJEllis committed:
Add script to automate grabbing+running node http tests

LewisJEllis committed:
Make instrumentation tests run on  Travis

LewisJEllis committed:
Drop 0.10 and 0.12 from instrumentation tests, make instr tests work on 4.x

LewisJEllis committed:
Update readme on instrumentation tests

LewisJEllis committed:
Add node-x.x.x dirs in instrumentation tests to gitignore

LewisJEllis committed:
Turn linter on for instrumentation tests

LewisJEllis committed:
Add restoreOriginals to breadcrumbs

LewisJEllis committed:
Add client.uninstall method

LewisJEllis committed:
Add some basic breadcrumb instrumentation tests

LewisJEllis committed:
Change breadcrumb url construction to depend on options passed instead of own parsed properties

LewisJEllis committed:
Add breadcrumb tests for basic capturre, sentry reqs, instrumenting missing module

LewisJEllis committed:
Add a couple comments on breadcrumb config and early exit behaviors

LewisJEllis committed:
Change default maxBreadcrumbs to 30

LewisJEllis committed:
Add context memory usage manual test

LewisJEllis committed:
Small cleanups to instrumentation test run.sh script

LewisJEllis committed:
Change unhandledRejection install flag to captureUnhandledRejections config flag

LewisJEllis committed:
Version 1.0.0

LewisJEllis committed:
Version 1.1.0; updated History.md

LewisJEllis committed:
Fix middleware backwards compatibility (#246)

* Fix middleware `client instanceof Raven.Client` check to use `Raven.constructor` (thanks @dhritzkiv)

* Add deprecation warning to top-level middleware

* Add tests for top-level Raven.middleware backwards compat

LewisJEllis committed:
Version 1.1.1

LewisJEllis committed:
Add 1.0.1 and 1.1.1 to History.md

LewisJEllis committed:
Docs 1.x overhaul (#249)

* Integration example docs

* working on context stuff

* WIP

* Added config stuff on breadcrumbs, cleaned up index overview

* Update usage docs with all the things

* Address Ben's feedback

LewisJEllis committed:
Fix integrations link in docs

Commit:
capture breadcrumbs from global context if not in active context

jeffwhelpley committed:
sending kwargs into shouldSendCallback() (#251)

LewisJEllis committed:
Fix koa doc variable name mixup, fixes #260 (#261)

LewisJEllis committed:
Add test for global breadcrumb capture

LewisJEllis committed:
Remove warning about getContext being called in global context

MaxBittker committed:
fixed lint and added seanbriceland's test

MaxBittker committed:
trim \r so tests pass on windows

MaxBittker committed:
different trim location

LewisJEllis committed:
Bind req/res to context domain in express requestHandler

LewisJEllis committed:
Add test for explicit adding req/res to context domain in middleware

LewisJEllis committed:
Make explicit binding test work on Node 0.10

MaxBittker committed:
check frame is inapp explicitly

LewisJEllis committed:
Fix postgres/pg name mismatch (#270)

LewisJEllis committed:
Add test for node core library frames being marked as not in_app (#271)

LewisJEllis committed:
Version 1.1.2

LewisJEllis committed:
Adds parseUser option to control user parsing behavior (#274)

LewisJEllis committed:
Add response flowing check to http instrumentation test

LewisJEllis committed:
Upgrade should devDep from ancient to 11.2

LewisJEllis committed:
Change http breadcrumb instrumentation to use emit monkeypatch instead of response listener

LewisJEllis committed:
Flowing state test compat for v0.10 (pre-streams3)

LewisJEllis committed:
Add alert about node vs raven-js when it seems like a browser env (#277)

benvinegar committed:
Link to Sentry grouping docs from fingerprint desc

LewisJEllis committed:
Version 1.1.3

mattrobenolt committed:
feat(transport): enable keep-alives and limit number of sockets

LewisJEllis committed:
Add contributing.md with release checklist/instructions

LewisJEllis committed:
Use util.format to get message string in console instrumentation

LewisJEllis committed:
Version 1.1.4

LewisJEllis committed:
Add sampleRate config option

LewisJEllis committed:
Add issue template

LewisJEllis committed:
Add sampleRate  tests

LewisJEllis committed:
Monkeypatch http.ClientReq#emit once instead of in each constructor

LewisJEllis committed:
Attach just original url to req object in http instrumentation

LewisJEllis committed:
Version 1.1.5

LewisJEllis committed:
Version 1.2.0

LewisJEllis committed:
Express patient app for manual memory tests

LewisJEllis committed:
Use child_process to have fresh express process for each test

benvinegar committed:
Allow manager.js to run from project root

benvinegar committed:
Add memwatch-next to dev deps

LewisJEllis committed:
Add README to manual memory tests

LewisJEllis committed:
Fix consoleAlert memory leak: unique messages no longer cached to print once

LewisJEllis committed:
Make manual memory test better isolate exception-capturing memory growth

LewisJEllis committed:
Update manual memory tests readme to reference prior memleak and mention warmup costs

LewisJEllis committed:
Remove memwatch-next from deps to make CI pass, add manual mem test running instructions

LewisJEllis committed:
Version 1.2.1

LewisJEllis committed:
Reorder release instruction steps

LewisJEllis committed:
Parse timeout from dsn qs param and timeout transport requests based on it

LewisJEllis committed:
global exception handler WIP

LewisJEllis committed:
Clean up fatal error handler and properly deal with recursion

LewisJEllis committed:
Exit condition tests

LewisJEllis committed:
Use new exception handling with domain error handlers and requestHandler

LewisJEllis committed:
Document fatal error handler behavior and expectations

LewisJEllis committed:
Use config field sendTimeout instead of DSN query param

LewisJEllis committed:
Document sendTimeout field

LewisJEllis committed:
Remove qs import from utils and timeout params from DSN tests

LewisJEllis committed:
Docs improvements on fatal error handler and timeout

LewisJEllis committed:
Make unhandledRejection handler only capture, not exit

LewisJEllis committed:
Properly set calledFatalError flag, add alert to onFatalError failsafe case

LewisJEllis committed:
Overhaul uncaught error handler to more reliably capture the first fatal exception before process shutdown

LewisJEllis committed:
Tests for exit behavior in face of multiple independent fatal errors in quuick succession

LewisJEllis committed:
Make default onFatalError callback console.error more defensive

LewisJEllis committed:
Add test for shutdown behavior with mixed domain + uncaughtException errors iin quick succession

LewisJEllis committed:
Add test for exception thrown after onFatalError is called

LewisJEllis committed:
Clean up fatal error handler docs wording

LewisJEllis committed:
Parse req from context, and requestHandler sets req on context

LewisJEllis committed:
Don't include domain prop as extra property on Error objects

LewisJEllis committed:
Add comment on req vs request naming, http interface, convenience, etc

LewisJEllis committed:
Add test for parsing req from context

LewisJEllis committed:
Remove redundant node version from travis

LewisJEllis committed:
Make .npmignore more strict for tidy installs

LewisJEllis committed:
Add req kwarg parsing docs, simplify req parsing in errorHandler

LewisJEllis committed:
Kill deprecated stuff for v2.0

LewisJEllis committed:
Version 2.0.0

LewisJEllis committed:
Make pg instrumentation a no-op

Commit:
use cache in parseStack, prevent unnecessary reading of files

LewisJEllis committed:
Add test for parseStack to read same file only once

LewisJEllis committed:
Refactor parseStack to break out file-reading and line-parsing into serial steps

LewisJEllis committed:
Add manual memory test for parseStack concurrency explosion on large module

LewisJEllis committed:
Remove npm upgrade from .travis.yml

LewisJEllis committed:
Version 2.0.1

LewisJEllis committed:
WIP: instrument require() for  autobreadcrumbs

LewisJEllis committed:
Made require hook work with https and uninstall

LewisJEllis committed:
Add failing test for sending empty req details when there's no req (#321)

LewisJEllis committed:
Only parse req property for request details if it is non-empty (fixes #321)

LewisJEllis committed:
Version 2.0.2

LewisJEllis committed:
Add test for require instrumentation double-hooking

LewisJEllis committed:
Fix double-hooking in require instrumentation

LewisJEllis committed:
Move fill to utils

LewisJEllis committed:
Add surrounding source truncation for long lines

LewisJEllis committed:
Add test for long line truncation

LewisJEllis committed:
Adds Node 8 to CI

LewisJEllis committed:
Fix additional error properties test for Node 8 error codes

MaxBittker committed:
Version 2.1.0 (#334)

Version 2.1.0
- Truncate long lines in surrounding source to avoid sending large amounts of minified code [See #329]
- Refactor automatic breadcrumb instrumentation of modules to accommodate compilation tools [See #322]
- Testing for Node 8 [See #328]

ambar committed:
delete kwargs.req after parsing

benvinegar committed:
Add test to verify `req` isn't appended to payload

benvinegar committed:
2.1.1

benvinegar committed:
2.1.2

kamilogorek committed:
Remove node archive after extracting

kamilogorek committed:
Update Node flags for v8.X instrumentation tests

Added flags:
--expose-internals
--expose-http2

Included one edge case, where we have to remove http2 flag.

kamilogorek committed:
Add logo to readme to streamline SDK readme (#367)

kamilogorek committed:
ref: Unify code style using Prettier and update ESLint rules

* Add Prettier, precommit hook using Husky and update ESLint to play nicely with Prettier.
* Run new code-style setup on all relevant .js files, without any changes in the code itself.
* Update ESLint version and unify package.json with the one from raven-js.

kamilogorek committed:
docs(support): Pre Node v4 deprecation

- wrote message about supported LTS versions
- removed `<v4` guard from instrumentation tests
- removed v0.10 and v0.12 from TravisCI config

kamilogorek committed:
Add test for maxReqQueueCount

kamilogorek committed:
Add test for maxReqQueueCount

kamilogorek committed:
feat(transport): retrieve socket name from http/s agents and add tests

kamilogorek committed:
fix: Pull Error's name from constructor, not Error itself

kamilogorek committed:
feat: Make private key in DSN optional

kamilogorek committed:
docs: Add Loopback integration docs

kamilogorek committed:
docs: Fix syntax error for Loopback integration

kamilogorek committed:
ref: Use local json-stringify-save package

kamilogorek committed:
feat: Add Errors serialization to json-stringify-safe

kamilogorek committed:
docs: Add name config description and optional ENV vars

kamilogorek committed:
docs: Include information about fatalError install callback

kamilogorek committed:
feat: Preserve some non-enumerable properties from request

kamilogorek committed:
ci: Trigger build only once and cache deps

kamilogorek committed:
docs: Add Sails.js integration docs

kamilogorek committed:
docs: Mention git-rev usage in release config attribute

kamilogorek committed:
Rewrote tests and simplified implementation

kamilogorek committed:
feat: Fall back to NODE_ENV for Sentry Environment

kamilogorek committed:
2.2.0

kamilogorek committed:
Remove memwatch-next dependency

kamilogorek committed:
fix: attach remaining non-enumerables to req

kamilogorek committed:
feat: Allow to configure stacktrace for captureMessage calls

kamilogorek committed:
docs: Add more clarification to req/request object

jorrit committed:
Fix typo in docs/config.rst

kamilogorek committed:
fix: access 'process' through global variable (#399)

kamilogorek committed:
ref: Warn user when using capture/wrap without installing Raven (#402)

kamilogorek committed:
ref: Enable http instrumentation by default (#401)

jan-auer committed:
ci: Integrate Zeus and release with the bot (#403)

* ci: Configure Zeus
* ci: Configure bot releases
* meta: Update contributing guidelines
* meta: Fix version bump docs

ali-essam committed:
feat: Support Boom statusCode

kamilogorek committed:
ref: Delete empty properties before sending event to the server (#407)

kamilogorek committed:
2.3.0

jan-auer committed:
meta: Make changelog compatible with release bot

anteriovieira committed:
docs: Update raven-js docs link (#409)

kamilogorek committed:
docs: Include Source Maps docs (#400)

kamilogorek committed:
fix: workaround for express.js 'host' deprecation message (#413)

kamilogorek committed:
feat: Sensible non-Error exception serializer (#416)

* feat: Sensible non-Error exception serializer
* feat: Use serialized keys for non-error messages
* test: Integration test for non-error ex serializer
* test: Fix non-errors exception tests for node < 6

kamilogorek committed:
2.4.0

kamilogorek committed:
docs: Add docs on how to use TypeScript SourceMaps (#419)

billyvg committed:
Update copyright message in LICENSE (#421)

kamilogorek committed:
docs: Added typescript reference in docs/index

oliviertassinari committed:
ref: Use a named function for the middleware over an anonymous (#424)

It's making the investigation of performance issues easier.

dustinblackman committed:
fix: Report fatal errors as indeed "fatal" (#425)

kamilogorek committed:
fix: Handle scoped packages structure in node-lsmod (#426)

* ref: Moved node-lsmod package to local vendor file
* fix: Handle scoped packages structure in node-lsmod

kamilogorek committed:
2.4.1

kamilogorek committed:
misc: Updated CONTRIBUTING.md with correct deploy steps

robhanlon22 committed:
docs: Update loopback docs (#427)

kamilogorek committed:
ref: Use named function for middlewares (#429)

urish committed:
fix: preserve context in for rejected promises (#428)

currently, context is lost in case of an unhandled promise rejection. This fix keeps track of the context by looking into the rejected promise domain. This fix has been tested in my production environment and seems to do the trick there.

rexxars committed:
fix: dont throw exception when called captureException without config phase

kamilogorek committed:
test: dont throw exception when called captureMessage without config phase

kamilogorek committed:
ref: Log promise rejection reason alongside eventid (#434)

kamilogorek committed:
2.4.2

timfish committed:
fix: Update stack-trace to handle spaces in paths (#437)

WanLinLin committed:
docs: Update Usage docs to include Domains in Promise support

WanLinLin committed:
docs: Update Usage docs to include Domains in Promise support (updated)

kamilogorek committed:
fix: Do not override context when capturing non-error exceptions

kamilogorek committed:
feat: log non-standard ports in breadcrumbs (#440)

kumavis committed:
feat: add flag for unhandled promise rejections

jan-auer committed:
docs: Remove private DSNs from the docs (#447)

evgenius committed:
fix: Remove a redundant try-catch block (#445)

kamilogorek committed:
2.5.0

kamilogorek committed:
fix: shallow-clone context instead of deep-copy (#452)

kamilogorek committed:
2.6.0

kamilogorek committed:
misc: Added package-lock.json

kamilogorek committed:
fix: correctly detect errors from vm (#457)

kamilogorek committed:
ref: use console.warn for alerts and store them in Set (#455)

* ref: use console.warn for alerts and store them in Set
* test: consoleAlert and consoleAlertOnce tests

kamilogorek committed:
ci: Add node 9 and 10 to travis builds (#456)

* ci: Add node 9 and 10 to travis builds

* test: Fix listeners test for Node >= 9

kamilogorek committed:
2.6.1

kamilogorek committed:
2.6.1 [fixed syntax error]

kamilogorek committed:
ref: remove redundant disableConsoleAlerts from tests

kamilogorek committed:
2.6.1 [fixed node10 tests]

jan-auer committed:
ref: Emit transaction instead of culprit (#458)

kamilogorek committed:
2.6.2

kamilogorek committed:
feat: capture exception breadcrumbs as we do in raven-js (#461)

kamilogorek committed:
misc: Repo archive note
